doc/directives.rst

```diff
@@ -74,8 +74,30 @@ adding the ``file`` option.
       :type: text
 
       The ``file`` option specifies to file to parse. The filename is
-      interpreted relative to the :data:`cautodoc_root` configuration
-      option. (For the time being, this option is mandatory.)
+      interpreted relative to the :data:`cautodoc_root` configuration option.
+      Each parsed file is cached for any following directive, but it is never
+      kept across reruns and it takes :rst:dir:`c:autodoc:clang` into account,
+      which means the file may still be parsed multiple times for a single
+      build.
+
+      This option may be omitted with some caveats:
+
+      1. It will be generally less performant than specifying it: when omitted,
+         a full cache will be built based on all files within
+         :data:`cautodoc_root` unless one was built by a previous directive
+         already. Omitting it in combination with bespoke per directive
+         :rst:dir:`c:autodoc:clang` options is particularly costly, potentially
+         triggering the parsing of all files for each and every directive.
+
+      2. If omitted, the variable ``name`` will be matched across the whole set
+         of cached entries that match the :rst:dir:`c:autodoc:clang` options.
+         Note that this includes any file outside the :data:`cautodoc_root`
+         search path that has been previously cached by a directive which
+         explicitly included it.
```

Comment:
Kept thinking about this. This last sentence is a lie in the current version... Only files in or nested `cautodoc_root` will ever be considered in this case, even if others were cached in between.